src/components/Hoverable/index.tsx

```diff
@@ -0,0 +1,213 @@
+import _ from 'lodash';
+import React, {MutableRefObject, ReactElement, RefAttributes, useCallback, useEffect, useImperativeHandle, useMemo, useRef, useState} from 'react';
+import {DeviceEventEmitter} from 'react-native';
+import * as DeviceCapabilities from '@libs/DeviceCapabilities';
+import CONST from '@src/CONST';
+import HoverableProps from './types';
+
+/**
+ * Maps the children of a Hoverable component to
+ * - a function that is called with the parameter
+ * - the child itself if it is the only child
+ * @param children The children to map.
+ * @param callbackParam The parameter to pass to the children function.
+ * @returns The mapped children.
+ */
+function mapChildren(children: ((isHovered: boolean) => ReactElement) | ReactElement | ReactElement[], callbackParam: boolean): ReactElement & RefAttributes<HTMLElement> {
+    if (Array.isArray(children) && children.length === 1) {
+        return children[0];
+    }
+
+    if (typeof children === 'function') {
+        return children(callbackParam);
+    }
+
+    return children as ReactElement;
+}
+
+/**
+ * Assigns a ref to an element, either by setting the current property of the ref object or by calling the ref function
+ * @param ref The ref object or function.
+ * @param element The element to assign the ref to.
+ */
+function assignRef(ref: MutableRefObject<HTMLElement> | ((element: HTMLElement) => void), element: HTMLElement) {
+    if (!ref) {
+        return;
+    }
+
+    if (typeof ref === 'function') {
+        ref(element);
+    } else if (_.has(ref, 'current')) {
+        // eslint-disable-next-line no-param-reassign
+        ref.current = element;
+    }
+}
```

Comment:
Remove lodash and improve param types:
```suggestion
function assignRef(ref: ((instance: HTMLElement | null) => void) | MutableRefObject<HTMLElement | null>, element: HTMLElement) {
    if (!ref) {
        return;
    }

    if (typeof ref === 'function') {
        ref(element);
    } else if (ref?.current) {
        // eslint-disable-next-line no-param-reassign
        ref.current = element;
    }
}
```